File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/Order/Ordering.cs
/// <filename>
///     Ordering.cs
/// </filename>
using System.Collections.Generic;
using Order;
using Alf.Library;
using Alf.Library.PrimitiveBehaviors;

namespace Order
{
    internal class Ordering : Order.IOrderManagementInterface
    {

        public Ordering()
        {
        }

        public virtual void addOrder(int productId, int amount)
        {
            OrderInfo order = new OrderInfo();
            order.productId = productId;
            order.amount = amount;
            this.orders.Add(order);
        }

        public virtual void removeOrder(int productId)
        {
            foreach (var order in this.orders)
            {
                if (order.productId == productId)
                {
                    this.orders.Remove(order);
                    break;
                }
            }
        }

        public virtual void orderProducts()
        {
            foreach (var order in this.orders)
            {
                this.dependency_ManageInventory.removeProductItems(order.productId, order.amount);
                string productName = this.dependency_ManageInventory.getProductName(order.amount);
                BasicInputOutput.WriteLine("Finish ordering " + IntegerFunctions.ToString(order.amount) + "x " + productName);
            }
            this.orders.Clear();
        }

        public System.Collections.Generic.List<OrderInfo> orders = new System.Collections.Generic.List<OrderInfo>();

        public Order.IInventoryInterface dependency_ManageInventory;
    }
}

File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/Order/Ordering.cpp
#include "Ordering.h"
#include <memory>
#include "OrderInfo.h"
#include "Ordering.h"
#include <vector>
#include <alf/library/collectionfunctions/CollectionFunctions.hpp>
#include "IInventoryInterface.h"
#include <alf/library/BasicInputOutput.hpp>
#include <string>

namespace Order
{
  Ordering::Ordering()
  {
  }
  void Ordering::addOrder(int productId, int amount)
  {
    std::shared_ptr<OrderInfo> order = std::make_shared<OrderInfo>();
    order->productId = productId;
    order->amount = amount;
    this->orders.push_back(order);
  }
  void Ordering::removeOrder(int productId)
  {
    for (auto& order : this->orders)
    {
      if (order->productId == productId)
      {
        alf::library::primitivebehaviors::CollectionFunctions::remove(this->orders, order);
        break;
      }
    }
  }
  void Ordering::orderProducts()
  {
    for (auto& order : this->orders)
    {
      this->dependency_ManageInventory->removeProductItems(order->productId, order->amount);
      std::string productName = this->dependency_ManageInventory->getProductName(order->amount);
      alf::library::BasicInputOutput::WriteLine(std::string("Finish ordering ") + std::to_string(order->amount) + std::string("x ") + productName);
    }
    this->orders.clear();
  }
}
